### ISSUE TITLE
docs build failing on obscure `@assert`

### ISSUE THREAD

**Issue by JeffFessler** [JuliaDocs/Documenter.jl#2384]
I have a Documenter build that is failing on this `@assert` in `parseblock`:
https://github.com/JuliaDocs/Documenter.jl/blob/d9865b4b73e4b0e37d242b65f77a87c1b7ffff4d/src/utilities/utilities.jl#L167

It seems quite hard to diagnose the cause of this error because `@assert` gives no context (other than the Stacktrace that simply says it originates in the `makedocs` call in `make.jl`).

Would you be open to a PR where we change the `@assert` to use a `throw` where info like the `file` could be shown as part of the error message?  That would help narrow down the search a lot.  Or other suggestions?

In case it helps, here is the build log:
https://github.com/JeffFessler/book-la-demo/actions/runs/7378584668/job/20073887407?pr=48

Thanks

**Comment by goerz** [comment]
The `@assert` isn't much different from a `throw` in that `@assert` can also emit a custom message with extra information. Try dev-installing `Documenter` and changing the relevant line 167 in `src/utilities/utilities.jl` to something like

```
@assert (length(exs) == 2 * length(results)) "Invalid code at $linenumbernode:\n$code"
```

These `@assert`'s are *internal* errors, though, so we should figure out what is happening here and then determine what the appropriate place for catching this with a helpful user-facing error message would be.

Documenter is running into some kind of parsing error while evaluating an `@example` block generated by Literate. If you can take a look at the `.md` files that `Literate` generates based on your demos, maybe you'll see something that sticks out.

**Comment by schillic** [comment]
This is probably a duplicate of #2206. In my case there was a space missing in a comment line, as in:
```julia
code...
#incorrect comment with missing leading space
# correct comment
code...
```

**Comment by JeffFessler** [comment]
Sorry for the duplicate, and thanks for the very helpful suggestions!  I will close this one then.

**Comment by JeffFessler** [comment]
> Try dev-installing `Documenter`

I took this suggestion and found that the culprit, which is a good reason to modify the assert so that it reports more information!

The issue is a paragraph consisting of two Julia comments like this:
```julia
## rank(Xr)
## svdvals(Xr)
```
with blank lines before and after.

These get translated into markdown like this:

```markdown
````@example lrmc-m
# rank(Xr)
# svdvals(Xr)
```

(There are four more backticks right after but I can't get github to show those.)
This looks like a perfectly legit example block to me (that happens to produce no output) so it seems like a Document bug.  As a workaround I will relocate the comments.

FWIW, earlier in my source there is a paragraph with a single Julia comment in it and that works fine.  So I wonder if this is a Documenter bug.

Compare:
https://github.com/JeffFessler/book-la-demo/blob/07b946abd4849bc69a9df1b0dc65a9fb42fbbea4/docs/lit/demos/10/lrmc-m.jl#L97

https://github.com/JeffFessler/book-la-demo/blob/07b946abd4849bc69a9df1b0dc65a9fb42fbbea4/docs/lit/demos/10/lrmc-m.jl#L186

I am reopening this because even if this is an internal error, I think the assert should be more descriptive.

**Comment by JeffFessler** [comment]
closed by #2388
